stdlib/universe/map_test.flux

```diff
@@ -538,39 +538,206 @@ testcase vectorize_nested_logicals {
 
     want =
         array.from(
             rows: [
                 {
-                    a: true,
-                    orAB: true,
-                    andAB: false,
-                    orBA: true,
-                    andBA: false,
-                    bothNullAnd: false,
-                    bothNullOr: false,
+                    nullOrFalse: debug.null(type: "bool"),
+                    falseOrNull: debug.null(type: "bool"),
+                    nullOrTrue: true,
+                    trueOrNull: true,
+                    nullAndFalse: false,
+                    falseAndNull: false,
+                    nullAndTrue: debug.null(type: "bool"),
+                    trueAndNull: debug.null(type: "bool"),
+                    nullAndNull: debug.null(type: "bool"),
+                    nullOrNull: debug.null(type: "bool"),
                 },
             ],
         )
+
+    got =
+        array.from(rows: [{_true: true, _false: false, _null: debug.null(type: "bool")}])
+            |> map(
+                fn: (r) =>
+                    ({r with
+                        nullOrFalse: r._null or r._false,
+                        falseOrNull: r._false or r._null,
+                        nullAndFalse: r._null and r._false,
+                        falseAndNull: r._false and r._null,
+                        nullOrTrue: r._null or r._true,
+                        trueOrNull: r._true or r._null,
+                        nullAndTrue: r._null and r._true,
+                        trueAndNull: r._true and r._null,
+                        nullAndNull: r._null and r._null,
+                        nullOrNull: r._null or r._null,
+                    }),
+            )
+            |> drop(columns: ["_null", "_true", "_false"])
+
+    testing.diff(want: want, got: got)
+}
+
+testcase logical_untyped_null_vectorized {
+    expect.planner(rules: ["vectorizeMapRule": 1])
+
+    want =
+        array.from(
+            rows: [{nullOrTrue: true, trueOrNull: true, nullAndFalse: false, falseAndNull: false}],
+        )
+            |> debug.opaque()
+
+    got =
+        array.from(rows: [{_true: true, _false: false}])
+            |> debug.opaque()
+            |> map(
+                fn: (r) =>
+                    ({r with
+                        nullOrFalse: r._null or r._false,
+                        falseOrNull: r._false or r._null,
+                        nullAndFalse: r._null and r._false,
+                        falseAndNull: r._false and r._null,
+                        nullOrTrue: r._null or r._true,
+                        trueOrNull: r._true or r._null,
+                        nullAndTrue: r._null and r._true,
+                        trueAndNull: r._true and r._null,
+                        nullAndNull: r._null and r._null,
+                        nullOrNull: r._null or r._null,
+                    }),
+            )
+            |> drop(columns: ["_true", "_false"])
+
+    testing.diff(want: want, got: got)
```

Comment:
At first I was confused how this test could pass then I realized the missing labels in the want where the values that should be null. Maybe adding a comment to remind the reader that a missing column is an untyped null would help?

Reply:
Added comments in 61f021a to explain how the untyped nulls are introduced. Yeah - it's a bit of a trick to make this happen on purpose :woozy_face: